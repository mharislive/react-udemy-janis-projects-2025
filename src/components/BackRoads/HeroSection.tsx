import Container from "./Container";
import main from "../../assets/backroads/main.jpeg";

const HeroSection = () => {
  return (
    <section
      id="hero-section"
      aria-label="Continue exploring"
      className="bg-gradient-to-b from-cyan-600/70 to-gray-500/70 bg-cover bg-center h-dvh relative overflow-hidden flex items-center justify-center"
    >
      <img
        src={main}
        alt="Hero"
        className="absolute left-1/2 -translate-x-1/2 top-1/2 -translate-y-1/2 w-auto h-full md:h-auto md:w-full -z-10"
      />
      <Container>
        <div className="flex flex-col gap-3 items-center text-white tracking-widest text-center">
          <h1 className="text-6xl uppercase font-bold">continue exploring</h1>
          <p className="max-w-[720px]">
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Vitae
            explicabo debitis est autem dicta.
          </p>
          <a
            href="#"
            className="text-xl px-6 py-3 text-cyan-500 hover:text-white bg-white hover:bg-transparent border border-white uppercase transition duration-500"
          >
            explore tours
          </a>
        </div>
      </Container>
    </section>
  );
};

export default HeroSection;
