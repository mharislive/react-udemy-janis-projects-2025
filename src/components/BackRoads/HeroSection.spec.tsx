import { render, screen } from "@testing-library/react";
import { axe } from "vitest-axe";
import HeroSection from "./HeroSection";

describe("BackRoads - Hero section", () => {
  it("should render successfully", () => {
    render(<HeroSection />);
    expect(screen.getByLabelText(/continue exploring/i)).toBeInTheDocument();
  });

  it("should render hero image", () => {
    render(<HeroSection />);
    expect(screen.getByAltText(/hero/i)).toBeInTheDocument();
  });

  it("should render heading", () => {
    render(<HeroSection />);
    expect(
      screen.getByRole("heading", { name: /continue exploring/i })
    ).toBeInTheDocument();
  });

  it("should render paragraph", () => {
    render(<HeroSection />);
    expect(screen.getByRole("paragraph")).toBeInTheDocument();
    expect(screen.getByRole("paragraph")).toHaveTextContent(/lorem/i);
  });

  it("should render link", () => {
    render(<HeroSection />);
    expect(
      screen.getByRole("link", { name: /explore tours/i })
    ).toBeInTheDocument();
  });

  it("should match snapshot", () => {
    const { asFragment } = render(<HeroSection />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should have no accessibility violations", async () => {
    const { container } = render(<HeroSection />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});
