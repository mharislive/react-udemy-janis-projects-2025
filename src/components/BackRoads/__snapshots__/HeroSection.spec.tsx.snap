// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BackRoads - Hero section > should match snapshot 1`] = `
<DocumentFragment>
  <section
    aria-label="Continue exploring"
    class="bg-gradient-to-b from-cyan-600/70 to-gray-500/70 bg-cover bg-center h-dvh relative overflow-hidden flex items-center justify-center"
    id="hero-section"
  >
    <img
      alt="Hero"
      class="absolute left-1/2 -translate-x-1/2 top-1/2 -translate-y-1/2 w-auto h-full md:h-auto md:w-full -z-10"
      src="/src/assets/backroads/main.jpeg"
    />
    <div
      class="container max-w-[1210px] mx-auto px-5 undefined"
    >
      <div
        class="flex flex-col gap-3 items-center text-white tracking-widest text-center"
      >
        <h1
          class="text-6xl uppercase font-bold"
        >
          continue exploring
        </h1>
        <p
          class="max-w-[720px]"
        >
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Vitae explicabo debitis est autem dicta.
        </p>
        <a
          class="text-xl px-6 py-3 text-cyan-500 hover:text-white bg-white hover:bg-transparent border border-white uppercase transition duration-500"
          href="#"
        >
          explore tours
        </a>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
